### DISCUSSION TITLE
Case Of Requiring Specific Number Of Egress Gateways

### DISCUSSION THREAD

**Discussion by infinitydon** [spidernet-io/egressgateway#1422] [Q&A]
Hi,

Is it possible to only want to create a specific number of GWs instead of leaving it wide open to using daemonsets? In some cases Egress IPs may be very limited for instance maybe 2 and the worker nodes are more than two, how do we create exact 2 GWs for the whole cluster?

**Comment by lou-lan** [comment]
Hi, I'm late here, I missed the GitHub discussion information. The EgressGateway is deployed using a `DaemonSet` because: The traffic from Pods to the Gateway Node uses a **VXLAN** tunnel, and the egressgateway-agent needs to establish a vxlan tunnel on each node.

If you want to avoid deploying Pods on all nodes, you can use node-selectors to achieve this.
